### ISSUE TITLE
AppProjects don't check all namespaces before finalizing deletion

### ISSUE THREAD

**Issue by sthomson-wyn** [argoproj/argo-cd#18361]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

If the resources finalizer is used on an AppProject, the application controller doesn't check all namespaces for existing Applications before finalizing deletion

**To Reproduce**

- create AppProject in default `argocd` namespace, with resources finalizer
- create Application in another namespace, under the aforementioned AppProject
- Delete the AppProject
- The Application is left hanging

**Expected behavior**

The AppProject waits for all Applications to delete before removing the finalizer

**Screenshots**

**Version**

```shell
argocd: v2.9.3+6eba5be.dirty
  BuildDate: 2023-12-02T00:36:55Z
  GitCommit: 6eba5be864b7e031871ed7698f5233336dfe75c7
  GitTreeState: dirty
  GoVersion: go1.21.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.10.2+fcf5d8c

```

The problem code appears to be here: https://github.com/argoproj/argo-cd/blob/ee9b38fe5ce15ce72ca57cdf16b0addbe2b0a27a/controller/appcontroller.go#L1015 only the default namespace is checked for Applications

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and tell us if the issue is still present, please?

**Comment by sthomson-wyn** [comment]
Yep it's still present https://github.com/argoproj/argo-cd/blob/b39f3b15fbf1abf59177b58b6e87a44e32600bbe/controller/appcontroller.go#L1070

**Comment by pazepaze** [comment]
The issue is still present in 3.0.12 when using "Applications in any namespace".

**Comment by pazepaze** [comment]
I think this duplicates #24347, which seems to have been fixed.
